unitTests/ui/ResultListTest.ts

```diff
@@ -832,59 +833,72 @@ export function ResultListTest() {
             expect(mockResultLayoutSelector.enableLayouts).toHaveBeenCalledWith(['card']);
           });
         });
-        describe('enableScrollToTop set to', () => {
-          let allResult: IQueryResults;
+
+        describe('scroll to top feature', () => {
           let scrollToTopSpy: jasmine.Spy;
-          const setupResultList = (
-            option: IResultListOptions = {
-              enableInfiniteScroll: true,
-              infiniteScrollContainer: document.createElement('div'),
-              enableScrollToTop: true
-            }
-          ) => {
-            let test = Mock.basicComponentSetup<ResultList>(ResultList, option);
-            test.cmp.currentlyDisplayedResults.push(allResult.results[1]);
-            (test.cmp.queryController.fetchMore as jasmine.Spy).and.returnValue(Promise.resolve(allResult));
+
+          const setupResultList = (options: Partial<IResultListOptions> = {}) => {
+            const test = Mock.advancedComponentSetup<ResultList>(ResultList, <Mock.AdvancedComponentSetupOptions>{
+              cmpOptions: {
+                infiniteScrollContainer: document.createElement('div'),
+                ...options
+              },
+              modifyBuilder: builder => {
+                return builder.withLiveQueryStateModel();
+              }
+            });
             return test;
           };
 
           beforeEach(() => {
             jasmine.clock().install();
             scrollToTopSpy = spyOn(ResultListUtils, 'scrollToTop');
-
-            allResult = FakeResults.createFakeResults(50);
           });
 
           afterEach(() => {
             jasmine.clock().uninstall();
           });
 
-          it('true, should call the scrollToTop of the ResultListUtils', done => {
-            test = setupResultList();
-            test.cmp.displayMoreResults(50).then(() => {
-              Simulate.query(test.env);
-              jasmine.clock().tick(0);
+          it(`when enableScrollToTop is true and enableInfiniteScroll is false
+          should scroll to top ONCE when page changes and a query is executed`, done => {
```

Comment:
Could this description be simplified to `...should scroll to top when the page changes.` I don't understand why `once` is included in caps?

Reply:
Because it should only be triggered once right after a QSM change for the "first" parameter, and not on subsequent requests